### PR TITLE
Initial converter

### PR DESCRIPTION
**Pull request by kzantow**
This PR implements the initial converter code, tests, simple workflow, and README.

### CODE REVIEW

**Review by wagoodman — COMMENTED** [review]
There are some bootstrapping things that should be addressed
- add bouncer checks to protect against future additions of dependencies that may have forbidden licenese
- adding golangci linter checks

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (wagoodman) [inline .gitignore:30]: we should include the same entries that are in the default golang gitignore for good measure https://github.com/github/gitignore/blob/main/Go.gitignore

File: converter.go [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+
+	if toTypePtr.Kind() != reflect.Ptr {
+		return fmt.Errorf("to struct provided was not a pointer, unable to set values: %v", to)
+	}
+
+	// toValue must be a pointer, but data needs to be set on the struct directly
+	toValue := toValuePtr.Elem()
+	toType := toValue.Type()
+
+	// Assume structs at this point
```
Comment (wagoodman) [inline converter.go]: nit: it would be a quick check to see if kind == struct and return an error

File: chain.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+// Convert converts from one type in the chain to the target type, calling each conversion in between
+func (c Chain) Convert(from interface{}, to interface{}) (err error) {
+	fromValue := reflect.ValueOf(from)
+	fromType := fromValue.Type()
+
+	// handle incoming pointers
+	for fromType.Kind() == reflect.Ptr {
+		fromValue = fromValue.Elem()
```
Comment (wagoodman) [inline chain.go:35]: clarification: is this resilient against passing nil here?

**Review by wagoodman — COMMENTED** [review]
optional suggestion: there are a lot of TODOs, which is alright, but since its early in this project it could be worth while capturing failing tests for some of the important cases that are left to do and segment them off with comments about the intended behavior. Or if not capture the failing test cases I think capturing these as issues would be helpful.

**Review by wagoodman — APPROVED** [review]
I really like it, awesome lib ❤️

File: converter.go (hunk 2 above)
Comment (kzantow) [reply converter.go]: This got me thinking, and in fact this wasn't necessary at all -- the `Convert` function can even handle primitives. I was able to remove some annoying duplicated code this way, too 👍

File: chain.go (hunk 3 above)
Comment (kzantow) [reply chain.go:35]: This _was_ a problem -- fixed and added tests

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      - name: Checkout
+        uses: actions/checkout@e2f20e631ae6d7dd3b768f56a5d2af784dd54791 # v2.5.0
+
+      - name: Setup Go
+        uses: actions/setup-go@c4a742cab115ed795e34d4513e2cf7d472deb55f # v3.3.1
+        with:
+          go-version: '1.18'
+
+      - name: Bootstrap tools
+        run: make bootstrap-tools
```
Comment (wagoodman) [inline .github/workflows/build.yml:16]: nit: could use some caching here, could come later too

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	$(LINTCMD) --fix
+	go mod tidy
+
+.PHONY: check-licenses
+check-licenses: ## Ensure transitive dependencies are compliant with the current license policy
+	$(TEMPDIR)/bouncer check ./...
+
+.PHONY: check-go-mod-tidy
+check-go-mod-tidy:
+	@ .github/scripts/go-mod-tidy-check.sh && echo "go.mod and go.sum are tidy!"
```
Comment (wagoodman) [inline Makefile]: I think this script is missing

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+SUCCESS := $(BOLD)$(GREEN)
+
+# test variables
+RESULTSDIR = test/results
+COMPARE_DIR = ./test/compare
+COMPARE_TEST_IMAGE = centos:8.2.2004
+COVER_REPORT = $(RESULTSDIR)/unit-coverage-details.txt
+COVER_TOTAL = $(RESULTSDIR)/unit-coverage-summary.txt
+# the quality gate lower threshold for unit test total % coverage (by function statements)
+COVERAGE_THRESHOLD := 80
```
Comment (wagoodman) [inline Makefile:27]: ❤️

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+CYAN := $(shell tput -T linux setaf 6)
+RED := $(shell tput -T linux setaf 1)
+RESET := $(shell tput -T linux sgr0)
+TITLE := $(BOLD)$(PURPLE)
+SUCCESS := $(BOLD)$(GREEN)
+
+# test variables
+RESULTSDIR = test/results
+COMPARE_DIR = ./test/compare
+COMPARE_TEST_IMAGE = centos:8.2.2004
```
Comment (wagoodman) [inline Makefile]: dead code

**Pull request merged**